src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs

```diff
@@ -472,7 +472,9 @@ private void SaveDocumentText(DocumentId id, string fullPath, SourceText newText
         {
             try
             {
-                var dir = Path.GetDirectoryName(fullPath);
+                var dir = Path.GetDirectoryName(fullPath) ?? Path.GetPathRoot(fullPath);
+                Contract.ThrowIfNull(dir);
```

Comment:
not sure i get if this is ok.  seems like a change in semantics./  @jasonmalinowski ?

Reply:
Reverted the change.